vyper/parser/stmt.py

```diff
@@ -481,7 +481,10 @@ def parse_assert(self):
             return LLLnode.from_list(['assert', test_expr], typ=None, pos=getpos(self.stmt))
 
     def _check_valid_range_constant(self, arg_ast_node, raise_exception=True):
-        arg_expr = Expr.parse_value_expr(arg_ast_node, self.context)
+        with self.context.range_scope():
```

Comment:
This seems like a really weird way to do this. Maybe @davesque has suggestions for alternatives?

Reply:
ContextManagers make sure you don't have to toggle the states by hand (i.e. have to keep track of all the possible exit points).

Reply:
Fair point. If it seems really clear to you, we can resolve this. I haven't been that deep lately to know better.

Reply:
It's the 'canonical' way to do it in this codebase, cf. https://github.com/ethereum/vyper/blob/9d14d017b369ec671fb959a65b98ccd9d5bdfe15/vyper/parser/context.py#L88-L117

In addition to being cleaner to read than toggling state by hand, it's also safer in the presence of exceptions (somebody may catch an exception at a higher level but without the `finally` clause the state will be corrupted).

Reply:
It was the recommended way to do this from https://github.com/ethereum/vyper/issues/1259